### ISSUE TITLE
Transactions on mocha-4 got stuck in the mempool

### ISSUE THREAD

**Issue by evan-forbes** [celestiaorg/celestia-app#2454]
When submitting some transaction using txsim, 4 managed to get stuck in the mempool, meaning that they were accepted by `CheckTx` but reject by `PrepareProposal`.

This should never happen, as malicious parties could fill the mempool with what appear to be valid high value transactions for free.

here are the transactions that got stuck

```json
{"jsonrpc":"2.0","id":-1,"result":{"n_txs":"4","total":"4","total_bytes":"4572","txs":["CqQBCqEBCiMvY29zbW9zLnN0YWtpbmcudjFiZXRhMS5Nc2dEZWxlZ2F0ZRJ6Ci9jZWxlc3RpYTEyMmhqZHdxdnE4bnQ3MmFrZnZzMmM0dHkzNmphNWRlcnJucnZ3dhI2Y2VsZXN0aWF2YWxvcGVyMTIyaGpkd3F2cThudDcyYWtmdnMyYzR0eTM2amE1ZGVyeHZwNGMyGg8KBHV0aWESBzk5NzkwMDASaApQCkYKHy9jb3Ntb3MuY3J5cHRvLnNlY3AyNTZrMS5QdWJLZXkSIwohAryvvomaK/5ZbWrhgd4YjZ4pN5SJa8IDLishkrv/DlMzEgQKAggBGAgSFAoOCgR1dGlhEgYyMTAwMDAQsJAfGkAVkaa/r1KUCwmaQuHFv+Vt3RwaBGChTlBsXTL3m2+0cRq5hJtrIAmniTqXzyS8LXBRchuI+Rjk4wzfhLQYQhEh","CuwCCukCCiovY29zbW9zLnN0YWtpbmcudjFiZXRhMS5Nc2dDcmVhdGVWYWxpZGF0b3ISugIKBwoFYWRtaXQSOwoSMTAwMDAwMDAwMDAwMDAwMDAwEhIyMDAwMDAwMDAwMDAwMDAwMDAaETEwMDAwMDAwMDAwMDAwMDAwGgcxMDAwMDAwIi9jZWxlc3RpYTFkZDhrOXhrOXFwbTV0OTUzdTdlazVnNWQ4a3FkN2VoOHJyZnM5Yyo2Y2VsZXN0aWF2YWxvcGVyMWRkOGs5eGs5cXBtNXQ5NTN1N2VrNWc1ZDhrcWQ3ZWg4eHV0Zm43MkMKHS9jb3Ntb3MuY3J5cHRvLmVkMjU1MTkuUHViS2V5EiIKIPNif9KxVmovBRf1kpvRCVH214WAMMkaKIwt+4j5Q65AOg8KBHV0aWESBzEwMDAwMDBCKjB4OWFGQzRFMUZGYTQ1OEE3MzAyOGJkRTNEODFkYUM0Njc1ZEEyOGYxZhJlCk4KRgofL2Nvc21vcy5jcnlwdG8uc2VjcDI1NmsxLlB1YktleRIjCiEC0pubVzDyX/z1uxhp2dZUXhTngeqJtErMU/hHtRdYEAMSBAoCCAESEwoNCgR1dGlhEgU4MDAwMBCUtCYaQKeehhN0WkVVN43GfXKZ8YnUHupDDCD1wTnOKl1iwQnuWScy4tQAjDdwJA6syXHqlYDiBArmYSUFSMqWgTlQndk=","CqACCqQBCjcvY29zbW9zLmRpc3RyaWJ1dGlvbi52MWJldGExLk1zZ1dpdGhkcmF3RGVsZWdhdG9yUmV3YXJkEmkKL2NlbGVzdGlhMXU1cHNodHFwZXhqbXV1ZHJ2cTZxMzM1cXltMnpnZ3pocDdrcTBwEjZjZWxlc3RpYXZhbG9wZXIxdTVwc2h0cXBleGptdXVkcnZxNnEzMzVxeW0yemdnemh5cDVlZTgKdwo7L2Nvc21vcy5kaXN0cmlidXRpb24udjFiZXRhMS5Nc2dXaXRoZHJhd1ZhbGlkYXRvckNvbW1pc3Npb24SOAo2Y2VsZXN0aWF2YWxvcGVyMXU1cHNodHFwZXhqbXV1ZHJ2cTZxMzM1cXltMnpnZ3poeXA1ZWU4EmcKUApGCh8vY29zbW9zLmNyeXB0by5zZWNwMjU2azEuUHViS2V5EiMKIQIi/tZFy5hp8aG0XFWnOyLh55CEq19GFSzvxDDC1X8ODRIECgIIARgQEhMKDQoEdXRpYRIFMjAwMDAQoJUJGkAM3O7HfrjgrlwxpxUM7hZuTxxY/hBhfMPmY/p7H38+H126fHXFOx8ExPpBT66EgXymC3aawg+Glm+87litewWm","CvQXCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFnM3hsd253ZGhhMnBxamh6cTR5ZGVnNGV2ZnZmdm03OGN6cHg3eBoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFkbXJ1NTN2aHlndTZjbjJjemQweWswbHRycXpwZ3p3amR5MDZtbRoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTF4dm0yd212cWxwMHpjNjB5N2N5dWM1NzdtdXUyYzVyOGZsemVtORoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFyZXoyODNkenZwOHg2cHd3ZjV1a3N5cjBxM2d6dzhzdXF0aGVlYRoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTE0bHdndHgwdDI2eXdkank4cDcweXVoeTVoeDg3cnY4ZTIyangzcBoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFzN2s2bjV4d3F4bnF0anMybHp5eTV6dnhmaHBjajU3eGwycWcwcxoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFzbXl3dXhhZXdtdHk1bTB5ZDg1czZ6bWQyN3FmbnptbjdnaDVjdxoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFjdG43OWhkeW1kenBlOGowMzg4cTA5Mmc5eXFwam0za2x4M3A0axoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTE4cTM1NzNqd3J4N2tlZWNucjJ4NHhubG00cGQwYzc0a3V4MDZkYRoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFmN3Fhanlwejh4ZWpseTRnN3M2bHdycGU0dWh3MDg4OXJqbTNjahoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFhZTk2M3ZsOXJmOTV6ajhoZHE5Y21jeW10MHlkYzh6dXY1eDU4aBoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFncWw0bGFwbXE2cGw1d3h3dGo4a3ZtMHB2MHh2eTV1amFxN3BrdRoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTE0MDA4YXdkanRrcXkwZTVnNmc1bHo0cGZ0YXY0dmpzYXRzN3R2bBoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFnNWE4OHdlY3F3dDV2ZGUwaHl5MnJlZ3I1dWg2NW42Z3BxOThrcxoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFmNjlhajI5bmwycW56anAzdXV3d3F0dHFkcHIwZTZxNmE3eThuNhoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTF1bnJlaHgzang0czBkdmFqd3B4a2Y5ZzdueWxqZ25tYTdkZWhmbhoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTF4NTVtemFmeG1ncGQ3OW53dWx1aHZsdWFtOWtqZ3BmNmgyMDVhNxoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTF2OHp0dGxtYXJobnFkdnhwbjhhbHZwZnAydXZucjdwbXlkMjh4ZhoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTF5MnplMDh2enJ3azkwMGthenQyZWVsOGR2MHByaHJqc2R3OWtlOBoSCgR1dGlhEgoxMDAwMDAwMDAwCpYBChwvY29zbW9zLmJhbmsudjFiZXRhMS5Nc2dTZW5kEnYKL2NlbGVzdGlhMW0yOHNxOXVyZjBud2p5NjhqY2ZsNTI2ZjJnemVoMzkwZmNtOTNsEi9jZWxlc3RpYTFyNnp4aHRmOHg2N2U0dHBhN200cHBwMGxoMDJteTZ0ZmFnenp4YxoSCgR1dGlhEgoxMDAwMDAwMDAwEmgKUApGCh8vY29zbW9zLmNyeXB0by5zZWNwMjU2azEuUHViS2V5EiMKIQPKyykhfXt/F3ieJC3ESaY+0rDl6QMQobBdjvosvR8pPhIECgIIARgbEhQKDgoEdXRpYRIGMjAwMDAwEICJehpAbjAq8ffHshcpgRFvpHcKQ2I3wyb1SIwqtBdrXpbQtcxKiVARMdcR1RWdQpaJL+Nsa7fkOnUzSRzuhPB80LOL/A=="]}}
```

**Comment by evan-forbes** [comment]
also, this was using txsim compiled from `main` I believe

**Comment by evan-forbes** [comment]
closing for now as this doesn't appear to be an issue after the patch to fix #2515

this is still something that we should look out for
